src/main/java/org/example/application/CountHealthInsuranceContributionService.java

```diff
@@ -0,0 +1,99 @@
+package org.example.application;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.example.domain.customer.Customer;
+import org.example.domain.customer.TaxPayments.FlatTax;
+import org.example.domain.customer.TaxPayments.GeneralTax;
+import org.example.domain.customer.TaxPayments.LumpSumTax;
+import org.example.domain.money.Currency;
+import org.example.domain.money.Money;
+import org.example.port.in.invoice.CountHealthInsuranceContributionUseCase;
+
+import java.math.BigDecimal;
+import java.time.YearMonth;
+
+@AllArgsConstructor
+@Getter
+
+public class CountHealthInsuranceContributionService implements CountHealthInsuranceContributionUseCase {
+    public static final Integer FIRST_CONTRIBUTION_RANGE = 5000;
+    public static final Integer SECOND_CONTRIBUTION_RANGE = 25000;
+    public static final BigDecimal MINIMUM_WAGE = BigDecimal.valueOf(3490);
+    private CountMonthlyMonthlyRevenueService countMonthlyRevenueService;
+    private CountMonthlyIncomeService countIncomeService;
+
+    public CountHealthInsuranceContributionService(CountMonthlyMonthlyRevenueService countMonthlyMonthlyRevenueService) {
+        this.countMonthlyRevenueService = countMonthlyMonthlyRevenueService;
+    }
+
+    @Override
+    public Money calculateHealthInsuranceContribution(Customer customer, YearMonth monthToCalculateHealthInsuranceContribution) {
+
+        if (LumpSumTax.class.equals(customer.getEntrepreneurshipForm().taxPaymentForm().getClass())) {
+            BigDecimal monthlyRevenue = countMonthlyRevenueService.countMonthlyRevenue(
+                            customer.getCustomerId(),
```

Comment:
to countMontlyMonthlyReven.. moglo by nalezec do Customera

Reply:
Yes, we will discuss it, I was thinking about before too. Provided the solution so it is the part of the customer implementation.